source/ConversationPanel.cpp

```diff
@@ -142,15 +143,23 @@ void ConversationPanel::Draw()
 		for(int side = 0; side < 2; ++side)
 		{
 			Point center = point + Point(side ? 420 : 190, 7);
+			Point unselected = point + Point(side ? 190 : 420, 7);
 			// Handle mouse clicks in whatever field is not selected.
 			if(side != choice)
 			{
 				AddZone(Rectangle(center, fieldSize), [this, side](){ this->ClickName(side); });
 				continue;
 			}
 			
-			// Fill in whichever entry box is active right now.
-			FillShader::Fill(center, fieldSize, selectionColor);
+			// Color selected text box, or flicker if user attempts an error
+			if(flickerTime % 6 > 3)
+				FillShader::Fill(center, fieldSize, dim);
+			else
+				FillShader::Fill(center, fieldSize, selectionColor);
```

Comment:
I'd simplify this with a ternary operator.
```suggestion
			FillShader::Fill(center, fieldSize, (flickerTime % 6 > 3) ? dim : selectionColor);
```